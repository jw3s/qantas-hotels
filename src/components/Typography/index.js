import React from 'react';
import styled from 'styled-components';

type HeadingProps = {
  as: 'h1' | 'h2' | 'h3' | 'h4',
  mb: number,
};
const headingSizeMap = {
  h1: 22,
  h2: 18,
  h3: 14,
  h4: 12,
};
const Heading = styled(({ as: As = 'h1', ...props }: HeadingProps) => <As {...props} />)`
  font-size: ${({ as = 'h1' }) => headingSizeMap[as] || 34}px;
  margin: 0;
  margin-bottom: ${({ mb = 20 }) => mb}px;
  color: #000;
`;

const Label = styled.label`
  font-size: ${({ size = 16 }) => size}px;
  font-weight: ${({ weight = 100 }) => weight};
  margin-bottom: ${({ mb = 5 }) => mb}px;
  color: ${({ color = 'initial' }) => color};
`;

export default {
  Heading,
  Label,
};
