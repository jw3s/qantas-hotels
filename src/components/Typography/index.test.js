import React from 'react';
import renderer from 'react-test-renderer';

import Typography from '.';

import 'jest-styled-components';

const { Heading, Label } = Typography;

describe('<Typography />', () => {
  describe('<Heading/>', () => {
    it('<Heading /> should render h1 tree', () => {
      const tree = renderer.create(<Heading>This is heading</Heading>).toJSON();
      expect(tree).toMatchSnapshot();
    });
    it('<Heading /> should render h2 tree', () => {
      const tree = renderer.create(<Heading as="h2">This is heading</Heading>).toJSON();
      expect(tree).toMatchSnapshot();
    });
    it('<Heading /> should render h3 tree with no margin', () => {
      const tree = renderer
        .create(
          <Heading as="h2" mb={0}>
            This is heading
          </Heading>,
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
  describe('<Label/>', () => {
    it('<Label /> should render simple label tree', () => {
      const tree = renderer.create(<Label>This is label</Label>).toJSON();
      expect(tree).toMatchSnapshot();
    });
    it('<Label /> should render label tree with 20px size weight bold', () => {
      const tree = renderer
        .create(
          <Label size={20} weight="bold">
            This is label
          </Label>,
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
