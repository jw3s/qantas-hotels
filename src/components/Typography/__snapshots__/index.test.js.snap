// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Typography /> <Heading/> <Heading /> should render h1 tree 1`] = `
.c0 {
  font-size: 22px;
  margin: 0;
  margin-bottom: 20px;
  color: #000;
}

<h1
  className="c0"
>
  This is heading
</h1>
`;

exports[`<Typography /> <Heading/> <Heading /> should render h2 tree 1`] = `
.c0 {
  font-size: 18px;
  margin: 0;
  margin-bottom: 20px;
  color: #000;
}

<h2
  className="c0"
>
  This is heading
</h2>
`;

exports[`<Typography /> <Heading/> <Heading /> should render h3 tree with no margin 1`] = `
.c0 {
  font-size: 18px;
  margin: 0;
  margin-bottom: 0px;
  color: #000;
}

<h2
  className="c0"
>
  This is heading
</h2>
`;

exports[`<Typography /> <Label/> <Label /> should render label tree with 20px size weight bold 1`] = `
.c0 {
  font-size: 20px;
  font-weight: bold;
  margin-bottom: 5px;
  color: initial;
}

<label
  className="c0"
  size={20}
>
  This is label
</label>
`;

exports[`<Typography /> <Label/> <Label /> should render simple label tree 1`] = `
.c0 {
  font-size: 16px;
  font-weight: 100;
  margin-bottom: 5px;
  color: initial;
}

<label
  className="c0"
>
  This is label
</label>
`;
